op-batcher/batcher/sync_actions_test.go

```diff
@@ -262,10 +262,10 @@ func TestBatchSubmitter_computeSyncActions(t *testing.T) {
 		},
 		{name: "no progress + unsafe=safe + blocks in state",
 			newSyncStatus: eth.SyncStatus{
-				HeadL1:    eth.BlockRef{Number: 5},
-				CurrentL1: eth.BlockRef{Number: 2},
-				SafeL2:    eth.L2BlockRef{Number: 101, Hash: block101.Hash()},
-				UnsafeL2:  eth.L2BlockRef{Number: 101},
+				HeadL1:      eth.BlockRef{Number: 5},
+				CurrentL1:   eth.BlockRef{Number: 2},
+				LocalSafeL2: eth.L2BlockRef{Number: 101, Hash: block101.Hash()},
+				UnsafeL2:    eth.L2BlockRef{Number: 101},
 			},
```

Comment:
Is it worth adding a test case which the old implementation would fail on? For example, the `LocalSafeL2 = UnsafeL2` and `SafeL2 < UnsafeL2` => `blocksToLoad = nil`.

Reply:
There would only be a noticeable difference in interop. Opened a ticket to track it: https://github.com/ethereum-optimism/optimism/issues/14376

In non-interop we'd have to craft a bad json-RPC response by modifying the op-node or creating a non-real unit test. The latter can be done, but not worthwhile I think. Open to adding it if you think it does help.